### PR TITLE
Add verify-golang-versions

### PR DESCRIPTION
**Pull request by sttts**
Validate golang version reference in different files to match.

```
include ./vendor/github.com/openshift/build-machinery-go/make/golang.mk

$(call verify-Dockerfile-builder-golang-version,images/Dockerfile)
$(call verify-go-mod-golang-version)
$(call verify-buildroot-golang-version)
$(call verify-golang-version-reference,some-custom-file,$(shell grep -e 'golang-[[:digit:]]*\.[[:digit:]]*' some-custom-file | sed 's/.*golang-//'))
```

or as the all-in-one variant:

```
$(call verify-golang-versions,images/Dockerfile)
```

### CODE REVIEW

File: make/targets/golang/version.mk [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+define add-Dockerfile-builder-golang-version
+$(call add-golang-version-reference,$(1),$(shell grep "AS builder" "$(1)" | sed 's/.*golang-\([[:digit:]][[:digit:]]*.[[:digit:]][[:digit:]]*\).*/\1/'))
+endef
+
+define add-go-mod-golang-version
+$(call add-golang-version-reference,go.mod,$(shell grep -e 'go [[:digit:]]*\.[[:digit:]]*' go.mod | sed 's/go //'))
+endef
+
+define add-buildroot-golang-version
+$(call add-golang-version-reference,.ci-operator.yaml,$(shell grep -e 'tag: golang-[[:digit:]]*\.[[:digit:]]*' .ci-operator.yaml | sed 's/.*tag: golang-//'))
```
Comment (mtrmac) [inline make/targets/golang/version.mk:50]: (The `grep | sed` can be combined by using `sed -n 's/…/…/p`, maybe using something like `sed -n 's/.*tag: golang-\([[:digit:]]*\.[[:digit:]]*.*\)/\1/p'`. That way the regex can’t get out of sync between the two statements, OTOH it’s using less known options.)

**Comment by EmilyM1** [comment]
This will be very helpful

File: make/examples/golang-versions-check/Makefile.test.log [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Golang version mismatch:
```
Comment (p0lyn0mial) [inline make/examples/golang-versions-check/Makefile.test.log:1]: this looks really good.

**Pull request merged**